### PR TITLE
Dobe/sysexpressions

### PR DESCRIPTION
**Pull request by dobe**

### CODE REVIEW

File: sql/src/main/java/io/crate/metadata/GlobalReferenceResolver.java [hunk 1]
```diff
@@ -30,11 +31,27 @@
     private final Map<ReferenceIdent, ReferenceImplementation> implementations;
 
     @Inject
-    public GlobalReferenceResolver(Map<ReferenceIdent, ReferenceImplementation> implementations) {
+    private NodeSysExpression nodeSysExpression;
```
Comment (mfelsche) [inline sql/src/main/java/io/crate/metadata/GlobalReferenceResolver.java]: i think this is never set

File: sql/src/main/java/io/crate/operation/projectors/SimpleTopNProjector.java [hunk 2]
```diff
@@ -72,6 +72,9 @@ public void startProjection() {
 
     @Override
     public synchronized boolean setNextRow(Object[] row) {
+        if (toCollect<1){
+            return false;
+        }
         assert toCollect >= 1;
```
Comment (mfelsche) [inline sql/src/main/java/io/crate/operation/projectors/SimpleTopNProjector.java]: this assertion can then be removed

File: sql/src/main/java/io/crate/metadata/GlobalReferenceResolver.java (hunk 1 above)
Comment (dobe) [inline sql/src/main/java/io/crate/metadata/GlobalReferenceResolver.java]: its injected

File: CHANGES.txt [hunk 3]
```diff
@@ -5,6 +5,13 @@ Changes for Crate
 Unreleased
 ==========
 
+ - The ``sys.operations`` table has now a ``_nodes`` system column
```
Comment (mfelsche) [inline CHANGES.txt]: sys.shards as well

**Comment by mfelsche** [comment]
otherwise ack

File: CHANGES.txt (hunk 3 above)
Comment (dobe) [inline CHANGES.txt]: this should not be official right now

**Pull request merged**